chemreg/compound/tests/test_models.py

```diff
@@ -45,6 +51,28 @@ def test_definedcompound_attr():
     assert inchikey.max_length == 29
 
 
+@pytest.mark.django_db(transaction=True)
+def test_query_structure_type():
+    """Test creation of `QueryStructureType` objects."""
+
+    # all the fields are non-nullable
+    with pytest.raises(IntegrityError):
+        QueryStructureTypeFactory(
+            name="ill-defined", label="Ill-Defined", short_description=None
+        )
```

Comment:
```suggestion
    for field in QueryStructureType._meta.get_fields():
        assert not field.null
        assert not field.blank
```

Example of non DB test.

Reply:
I have kept the database tests in place because I had not refreshed your comments. I'll take another look

Reply:
A test like this confirms that we've specified the model a certain way, but it doesn't test the functional outcomes of those specifications. It seems worth the trouble to actually make sure that the serializer rejects a null field, rather than just checking whether the field is set to nullable.